gips/0054-timeline-aggregation-protocol.md

```diff
@@ -0,0 +1,270 @@
+[2023.03 TAP security.pdf](../assets/gip-0054/2023.03_tap_security.pdf)
+
+Finally, the protocol and the reference implementations are open source. Abiding by Kerckoffs’s principle and providing the opportunity for the protocol to be further improved by the community. 
+
+
+
+# Validation
+
+We got the protocol reviewed by an external consultant. Details can be found in:
+[2023.03 TAP Discussion (External Review).pdf](../assets/gip-0054/2023.03_tap_discussion_external_review.pdf)
+
+We will also schedule an audit of all the implemented smart contracts before their deployment.
+
+# Rationale and Alternatives
+
+The journey to Graph-scale payments is storied and branched into several potential solution spaces. What follows is an abbreviated history of things we tried and why they didn't work.
+
+## Other state channel systems
+Before creating Scalar, we attempted to integrate several pre-existing or in-development state channel systems, including but not limited to Vector, Nitro, and Indra. Although each system had unique tradeoffs, each exhibited one or more design flaws that manifested in a total system breakdown at scale.
+
+### Slow, serial channel creation
+The ability to create channels quickly is desirable when the system is starting up, cycling allocations, handling long-running requests, or experiencing unexpected load. The systems we evaluated throttled state channel creation, usually by requiring participants to update a shared data structure (such as a Merkelized root of outstanding channels). The result of high-latency, low-throughput state channel creation ranged from degraded quality of service to denial-of-service attack vulnerability.
+
+### Inefficient use of capital 
+The systems we evaluated allocated capital to individual state channels rather than the state channel pool. Resource partitioning is generally wasteful. And in our case, the wasted cost of capital increased the system's friction.
+
+### Fragile failure modes
+The systems we evaluated were designed around the happy path, with failure modes as an afterthought. In a distributed system, however, "exceptional cases" are a common occurrence. Rather than degrading safely to a reasonable state, some systems would get stuck and require human intervention due to what should have been innocuous events like network failures or race conditions.
+
+### Trust Assumptions
+Scalar TAP reduces the amount of trust required between participants. But, even the previous, more trusted version of Scalar had lower trust requirements than the state channel systems we evaluated. The trust problem that surfaces when The Graph uses state channels is that with many small state channels, collecting the channels in a dispute can cost more than the total value locked in the channels. That means that the participants are assuming collaboration, and in the worst case, the system was effectively a fancy wrapper over asking people nicely to pay at the end.
```

Comment:
It might be helpful to specifically articulate the trust assumptions under Scalar TAP.

Reply:
good suggestion, maybe I will ask Zac to give it a shot (not sure why I cannot even mention him in here... ❓